### PR TITLE
adding new method gettingHistoryArray and trying to update history div

### PR DESCRIPTION
**Pull request by yaaHosein**
let us work together!!

### CODE REVIEW

File: DevelopedCalculator.js [hunk 1]
```diff
@@ -13,14 +13,20 @@ class DevelopedCalculator extends Calculator {
...
             this.hostElem.prepend(this.historyDiv);
+            //this.historyDiv.gettingHistoryArray();
         }
         // new method has a parameter
     onButtonNumberClick(event) {
         // calling parent class's onButtonNumberClick method
         super.onButtonNumberClick(event);
         // pushing the history array with current result
         this.history.push(this.currentResult);
+        //this.historyDiv.gettingHistoryArray();
```
Comment (SMH110) [inline DevelopedCalculator.js]: can you explain what `historyDiv` is?

Comment (yaaHosein) [reply DevelopedCalculator.js]: It is the div element where we can seek the history in it

Comment (SMH110) [reply DevelopedCalculator.js]: do `div` elements have `gettingHistoryArray`  method?

Comment (yaaHosein) [reply DevelopedCalculator.js]: No they do not 
I have pushed some updates before reading this comment

File: DevelopedCalculator.js [hunk 2]
```diff
@@ -2,25 +2,30 @@
...
         // new method has a parameter
     onButtonNumberClick(event) {
         // calling parent class's onButtonNumberClick method
         super.onButtonNumberClick(event);
         // pushing the history array with current result
         this.history.push(this.currentResult);
+        this.gettingHistoryArray();
     }
+    gettingHistoryArray() {
+        this.history.join(",");
```
Comment (SMH110) [inline DevelopedCalculator.js]: `this.history.join(",");` returns the history items separated by ","
I can't see your method `gettingHistoryArray` doing anything useful.
despite this part of the code ``this.history.join(",");` is correct and needed, but you're not using that in anything 

you're not updating the historyDiv which you renamed to historyElement with returned value of `this.history.join(",");

can you read the requirements again pls?

**Comment by yaaHosein** [comment]
set the history content which is separated by comma as content of the history div
for this you need to access the history div and update its content.

I have tried to add this code:
this.historyElement.gettingHistoryArray();
Inside render body 
But I got error message in console tab

**Comment by SMH110** [comment]
> set the history content which is separated by comma as content of the history div
> for this you need to access the history div and update its content.
> 
> I have tried to add this code:
> this.historyElement.gettingHistoryArray();
> Inside render body
> But I got error message in console tab

I am going to help you by asking you questions because we covered something similar before.

You need to call gettingHistoryArray in method that responsible of updating the result element.
Is render method called when a button clicked?
If yes then you are doing good. But if not then calling gettingHistoryArray in the render method won't work.


The second question.
You told me before (today) when i asked about the historyDiv ( now it's called historyElement)
"It is the div element where we can seek the history in it"
So my question how to update a div content?

File: DevelopedCalculator.js [hunk 3]
```diff
@@ -25,6 +24,7 @@ class DevelopedCalculator extends Calculator {
     }
     gettingHistoryArray() {
         this.history.join(",");
+        this.historyElement.textContent = this.currentResult;
```
Comment (SMH110) [inline DevelopedCalculator.js]: This part `this.historyElement.textContent =` is correct 
But you are assigning the wrong value.
Do we need the current result or the history content?

Comment (yaaHosein) [reply DevelopedCalculator.js]: Maybe I pushed the updates in the same time of pushing your comment

File: DevelopedCalculator.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ class DevelopedCalculator extends Calculator {
     }
     gettingHistoryArray() {
         this.history.join(",");
-        this.historyElement.textContent = this.currentResult;
+        this.historyElement.textContent = this.history;
```
Comment (SMH110) [inline DevelopedCalculator.js]: `this.history` is the array of the clicked button,
I don't know what is the point of having this ` this.history.join(",");` if you're not using it?

**Comment by yaaHosein** [comment]
Unfortunately I do not have much of experience to work with arrays 
And do not at all where and how I should use it 

I checked to frame 
this.history in line 27 by []

I did not get error message in console tab

Also still now Do not know what is the reason behind using arrays 

Also in other previous months we had result div and result value 
But never used arrays 
So please what’s the new thing which is making us to use arrays 

Thanks

**Comment by SMH110** [comment]
> Unfortunately I do not have much of experience to work with arrays
> And do not at all where and how I should use it
> 
> I checked to frame
> this.history in line 27 by []
> 
> I did not get error message in console tab
> 
> Also still now Do not know what is the reason behind using arrays
> 
> Also in other previous months we had result div and result value
> But never used arrays
> So please what’s the new thing which is making us to use arrays
> 
> Thanks

Arrays in javascript allow you to store more than one value in a single variable.
If I want to store more than one value, i can use array to do that 

say for example John likes three colours blue, red and yellow 
if I want to write that in js 
I can do

```js 
const  johnFavourteColor1 = "blue"
const johnFavourteColor2  = "red"
const johnFavourteColor3 = "yellow"
```

or I can do something better 
```js
const johnFavourteColors = ["blue", "red", "yellow"]
```

Arrays in javascript are indexed and the start index is zero   
`const foo = ["z"]` 

If I do 
`foo[0]`  I get "z" 
Arrays in javascript has length property which returns the length of the array (how many items contain)
Arrays in javascript has methods like `push` which push item to the end of the array 
`pop` removes the last item of the array and return it 
`shift` removes the first item of the array and return it

The reason we are using array for the history, because history can be many while current result is only one result 

I said this many time  and I will repeat 

If I click on these buttons 

1 + 2 + 3 + 4 = 
I see  10 in the result element 
but I want to see   1, 2, 3, 4 in the history element 
I want to see that 10 which I am seeing in the result div from which numbers (buttons) came 
 
 I want to see how many buttons I clicked in order I got the 10 in the result element 

that's why everytime a button clicked,  the value of that button should be pushed to the history array 
so the history array would look like 
[1, 2, 3, 4]

**Comment by yaaHosein** [comment]
Thank you so much for re-clearing the reason 
But I never saw in any calculator the history separated by ,
Usually it is separated by operator signs 
1-2+4*3-5+6/5=

So why you are joining , between numbers 

I know the values in an array should be separated by ,  
But the history which will appear for me should not separated by , usually 
Thanks

**Comment by SMH110** [comment]
> Thank you so much for re-clearing the reason
> But I never saw in any calculator the history separated by ,
> Usually it is separated by operator signs
> 1-2+4*3-5+6/5=
> 
> So why you are joining , between numbers
> 
> I know the values in an array should be separated by ,
> But the history which will appear for me should not separated by , usually
> Thanks

If you understood what I have said, I want you to add a comments to explain everything in detail 
please try to not copy my words.

in the future will add the operators only, I don't want to confuse you from now 
I wanted to make it easier for you

**Comment by yaaHosein** [comment]
Ok thanks 
InshaAllah 
Also I wasn’t asking you to explain arrays in JavaScript for me again 

Just My question was because i was confused for using arrays 
And you re explained it thankfully

File: DevelopedCalculator.js [hunk 5]
```diff
@@ -20,10 +20,12 @@ class DevelopedCalculator extends Calculator {
         super.onButtonNumberClick(event);
         // pushing the history array with current result
         this.history.push(this.currentResult);
+        // calling gettingHistoryArray
```
Comment (SMH110) [inline DevelopedCalculator.js]: It wil lbe good to add in the comments the reason of calling `this.gettingHistoryArray`

File: DevelopedCalculator.js [hunk 6]
```diff
@@ -2,25 +2,31 @@
...
         // calling parent class's onButtonNumberClick method
         super.onButtonNumberClick(event);
         // pushing the history array with current result
         this.history.push(this.currentResult);
+        // calling gettingHistoryArray to update the history element after clicking any number button 
+        this.gettingHistoryArray();
     }
+    gettingHistoryArray() {
+        // updating the history element by the history which is joined with the clicked operators in the future (.. will be changed with an operator)
+        this.historyElement.textContent = this.history.join("..");
```
Comment (SMH110) [inline DevelopedCalculator.js]: Can you use comma ","  as separator instead of ".." ?

**Comment by SMH110** [comment]
**Next Task**
You have implemented pushing the current result to the history array when a button clicked.
If we wanted to do **something similar** to the **operators** buttons what should we do?

what need to be done to show the operator in the history element.
for example if I click on these buttons 
1 + 2 + 3

at the moment I can see this in the hisotry element
`1..12..123.. `

what need to be done to see 
1..+..12..+123..  ?

**Comment by SMH110** [comment]
I only asked a question, 
can you please tell me what need to be done?

1.
2.
3.
..etc

**Comment by yaaHosein** [comment]
> I only asked a question,
> can you please tell me what need to be done?
> 
> 
> ..etc

1- need to create new method for operator buttons and it’s better to has the same name in the parent Calculator class 
2- we need to add inside this method same codes which added in 
On number buttons click method 
But by changing number to operator 
3. We need to address the line which register the operator buttons when they be checked 
I am not near the laptop now to write it 
It has target in it
this.operator= 


4. We need to update the gettingArrayHistory method by joining .. + this.operator

**Comment by SMH110** [comment]
Please think again and tell me what need to be done in the new method which you mentioned when you said "create new method for operator buttons"

Please make sure that your answer does not mention `gettingArrayHistory ` because this method should not change anymore

**Comment by yaaHosein** [comment]
> Please think again and tell me what need to be done in the new method which you mentioned when you said "create new method for operator buttons"
> 
> Please make sure that your answer does not mention `gettingArrayHistory ` because this method should not change anymore

ok
thanks

**Comment by yaaHosein** [comment]
> > Please think again and tell me what need to be done in the new method which you mentioned when you said "create new method for operator buttons"
> > Please make sure that your answer does not mention `gettingArrayHistory ` because this method should not change anymore
> 
> ok
> thanks

1- calling parent class’s onOperatorButtonClick method 
2. Adding the clicked operator

**Comment by SMH110** [comment]
> > > Please think again and tell me what need to be done in the new method which you mentioned when you said "create new method for operator buttons"
> > > Please make sure that your answer does not mention `gettingArrayHistory ` because this method should not change anymore
> > 
> > 
> > ok
> > thanks
> 
> 1- calling parent class’s onOperatorButtonClick method
> 2. Adding the clicked operator

Please undo the recent changes you did.
After that do not push anything.

Can you explain "Adding the clicked operator"

**Comment by yaaHosein** [comment]
> > > > Please think again and tell me what need to be done in the new method which you mentioned when you said "create new method for operator buttons"
> > > > Please make sure that your answer does not mention `gettingArrayHistory ` because this method should not change anymore
> > > 
> > > 
> > > ok
> > > thanks
> > 
> > 
> > 1- calling parent class’s onOperatorButtonClick method
> > 2. Adding the clicked operator
> 
> Please undo the recent changes you did.
> After that do not push anything.
> 
> Can you explain "Adding the clicked operator"

I mean choosing the clicked operator - || +||*||/

Thanks

**Comment by SMH110** [comment]
> I mean choosing the clicked operator - || +||*||/

if I clicked on these buttons 

1 + 2 - 1

here I have two operators:
1. +
2. -

and I want to see them in the history element 
1,+, 2,-,1

how "choosing the clicked operator - || +||*||/" will work here?
we are dealing with more than one operator

**Comment by yaaHosein** [comment]
Yes you are right therefore I wrote for you 
-||+||-||*||/

So if click on/ for example I will see /

If click * I will see * in history 

And so on 

Thanks

**Comment by SMH110** [comment]
> Yes you are right therefore I wrote for you
> -||+||-||*||/
> 
> So if click on/ for example I will see /
> 
> If click * I will see * in history

I just copied this `-||+||-||*||/` and pasted in the Console tab in the developer tools.
this is what I got 
image

so I wrapped each operator with quote like this
`"-" || "+" || "*" || "/"` 
and the result of the code above 
image

this means everytime you run this code `"-" || "+" || "*" || "/"` 
you will get the same and the same and the same and the same and the same result 
you will always get "-" 

so If I click on + , will I see "-" in the history element?

**Comment by SMH110** [comment]
If I tell you that we want to perform some operations on 1000 different numbers
like 
1 + 2 - 3 + 54 * 100  / 6 - 100 ...etc
before I start reading the numbers, I ask you to remember all the operators I am going to mention and in the order I will say them.
what you will do?

**Comment by yaaHosein** [comment]
Sorry I didn’t not write note for you 
I never meant by using this sign || 
Its use in JavaScript 

I mean by this sign || or

Also when I pushed the codes to the project I did not use ||

**Comment by yaaHosein** [comment]
> If I tell you that we want to perform some operations on 1000 different numbers
> like
> 1 + 2 - 3 + 54 * 100 / 6 - 100 ...etc
> before I start reading the numbers, I ask you to remember all the operators I am going to mention and in the order I will say them.
> what you will do?

I can not memorize these so much operators

**Comment by SMH110** [comment]
> > If I tell you that we want to perform some operations on 1000 different numbers
> > like
> > 1 + 2 - 3 + 54 * 100 / 6 - 100 ...etc
> > before I start reading the numbers, I ask you to remember all the operators I am going to mention and in the order I will say them.
> > what you will do?
> 
> I can not memorize these so much operators

how can you make this possible? I don't mind if you use any thing to make this possible

**Comment by SMH110** [comment]
> Sorry I didn’t not write note for you
> I never meant by using this sign ||
> Its use in JavaScript
> 
> I mean by this sign || or
> 
> Also when I pushed the codes to the project I did not use ||

Show me here in the comments in one line what you mean

**Comment by yaaHosein** [comment]
> > Sorry I didn’t not write note for you
> > I never meant by using this sign ||
> > Its use in JavaScript
> > I mean by this sign || or
> > Also when I pushed the codes to the project I did not use ||
> 
> Show me here in the comments in one line what you mean

2. Adding the clicked operator
- or + or / or *

**Comment by SMH110** [comment]
> > > Sorry I didn’t not write note for you
> > > I never meant by using this sign ||
> > > Its use in JavaScript
> > > I mean by this sign || or
> > > Also when I pushed the codes to the project I did not use ||
> > 
> > 
> > Show me here in the comments in one line what you mean
> 
> 1. Adding the clicked operator
> 
> * or + or / or *

show me code please
I want to see how you're going to translate "Adding the clicked operator" to a code
because you used the word "adding" which means you want to add something to another something 
but I don't know what those somethings are ?!

**Comment by yaaHosein** [comment]
> > If I tell you that we want to perform some operations on 1000 different numbers
> > like
> > 1 + 2 - 3 + 54 * 100 / 6 - 100 ...etc
> > before I start reading the numbers, I ask you to remember all the operators I am going to mention and in the order I will say them.
> > what you will do?
> 
> I can not memorize these so much operators

Firstly how you can tell me them without telling me the numbers 
Secondly I will take a piece of paper and write them

**Comment by SMH110** [comment]
> Firstly how you can tell me them without telling me the numbers


 I am going to read the numbers with the operators, so I would say 
one plus two minus five ...etc 


> Secondly I will take a piece of paper and write them
This is a good answer, but can you show me what you would write on the paper if I said

1 + 2 - 6 * 10 

+ - *
Plus subtract multiply

**Comment by yaaHosein** [comment]
> > > > Sorry I didn’t not write note for you
> > > > I never meant by using this sign ||
> > > > Its use in JavaScript
> > > > I mean by this sign || or
> > > > Also when I pushed the codes to the project I did not use ||
> > > 
> > > 
> > > Show me here in the comments in one line what you mean
> > 
> > 
> > 
> > 1. Adding the clicked operator
> > 
> > 
> > * or + or / or *
> 
> show me code please
> I want to see how you're going to translate "Adding the clicked operator" to a code
> because you used the word "adding" which means you want to add something to another something
> but I don't know what those somethings are ?!

Yes adding that line of code 
This.operator = ....
To the mentioned method

**Comment by yaaHosein** [comment]
> > > > > Sorry I didn’t not write note for you
> > > > > I never meant by using this sign ||
> > > > > Its use in JavaScript
> > > > > I mean by this sign || or
> > > > > Also when I pushed the codes to the project I did not use ||
> > > > 
> > > > 
> > > > Show me here in the comments in one line what you mean
> > > 
> > > 
> > > 
> > > 1. Adding the clicked operator
> > > 
> > > 
> > > * or + or / or *
> > 
> > 
> > show me code please
> > I want to see how you're going to translate "Adding the clicked operator" to a code
> > because you used the word "adding" which means you want to add something to another something
> > but I don't know what those somethings are ?!
> 
> Yes adding that line of code
> This.operator = ....
> To the mentioned method

I have pushed the code also 
It’s start with 
this.operator=

**Comment by SMH110** [comment]
> > > > > > Sorry I didn’t not write note for you
> > > > > > I never meant by using this sign ||
> > > > > > Its use in JavaScript
> > > > > > I mean by this sign || or
> > > > > > Also when I pushed the codes to the project I did not use ||
> > > > > 
> > > > > 
> > > > > Show me here in the comments in one line what you mean
> > > > 
> > > > 
> > > > 
> > > > 1. Adding the clicked operator
> > > > 
> > > > 
> > > > * or + or / or *
> > > 
> > > 
> > > show me code please
> > > I want to see how you're going to translate "Adding the clicked operator" to a code
> > > because you 
> I have pushed the code also
> It’s start with
> this.operator=
Can you remind how the this.operator is used.
Why did we add this property.
Also this.operator = ????????
How this is equal to  "adding the clicked operator"

**Comment by SMH110** [comment]
This is very good.

> Plus subtract multiply

If i wrap these three items with [ ] it will almost look like js array
[Plus subtract multiply]

I can tell that you will write Plus then you will write subtract  and you will not write subtract on top of Plus
You will add subtract next to Plus.

But why when it come to js you do the opposite?

You were doing something like

this.operator = ...etc
 Why not using the history array to push the operator 
This why you will add subtract next to Plus not on top of it

**Comment by yaaHosein** [comment]
> This is very good.
> 
> > Plus subtract multiply
> 
> If i wrap these three items with [ ] it will almost look like js array
> [Plus subtract multiply]
> 
> I can tell that you will write Plus then you will write subtract and you will not write subtract on top of Plus
> You will add subtract next to Plus.
> 
> But why when it come to js you do the opposite?
> 
> You were doing something like
> 
> this.operator = ...etc
> Why not using the history array to push the operator
> This why you will add subtract next to Plus not on top of it

Yes you are absolutely  right 
Using the wrong codes was because of not having much experience in working with arrays 
And aside from forgetting the role of 
super.onOperatorButtonClick
Which has the role of adding the clicked operator

**Comment by yaaHosein** [comment]
Please a question 
If I click 1-
I will see 1 but I will not see the -

Is - a part of current result?!

I am not asking about history element and child class

File: DevelopedCalculator.js [hunk 7]
```diff
@@ -30,8 +30,8 @@ class DevelopedCalculator extends Calculator {
     onOperatorButtonClick(event) {
         // calling parent class's onOperatorButtonClick method
         super.onOperatorButtonClick(event);
-        // for adding the clicked operator 
-        this.operator = event.target.dataset.operator;
+        // pushing the history array with current result
```
Comment (SMH110) [inline DevelopedCalculator.js]: what do you mean by " pushing the history array with current result"

you're pushing the "currentResult" to the history array 
is this correct?

**Comment by SMH110** [comment]
> And aside from forgetting the role of
> super.onOperatorButtonClick

I don't know what to say about this, I asked you to add comments before to help yourself.

**Comment by SMH110** [comment]
# Question
How do you figure out which operator was clicked in the `onOperatorButtonClick` ?

**Comment by yaaHosein** [comment]
> # Question
> How do you figure out which operator was clicked in the `onOperatorButtonClick` ?

1. (Long journey) Logging event
Clicking the operator button
Opening mouse event 
Opening target 
Hovering the mouse pointer on button.button
Or clicking on it 

2. ( short journey)
Logging this.currentResult 

I know why you are asking me this question 
I got my question’s answer 
Thanks

**Comment by yaaHosein** [comment]
Unfortunately the comment was in front of me but in the onButtomNumberClick firstly

**Comment by SMH110** [comment]
> 2. ( short journey)
>    Logging this.currentResult


Please show me, 
log the ` this.currentResult` inside `onButtomNumberClick ` 

then click on button + and show me that you get "+" or something similar in the console 
send me a screenshot pls

**Comment by yaaHosein** [comment]
> # Question
> How do you figure out which operator was clicked in the `onOperatorButtonClick` ?

1. (Long journey) Logging event
Clicking the operator button
Opening mouse event 
Opening target 
Hovering the mouse pointer on button.button
Or clicking on it 

2. ( short journey)
Logging this.currentResult 

I know why you are asking me this question 
I got my question’s answer 
Thanks

**Comment by SMH110** [comment]
> Question
> How do you figure out which operator was clicked in the onOperatorButtonClick ?
> 
> (Long journey) Logging event
> Clicking the operator button
> Opening mouse event
> Opening target
> Hovering the mouse pointer on button.button
> Or clicking on it
> 
> ( short journey)
> Logging this.currentResult
> 
> I know why you are asking me this question
> I got my question’s answer
> Thanks

I can't see the screenshot, maybe you took very big screenshot 
you can use snipping tools to take screenshot 
then copy (ctrl + c) and paste in your comment

**Comment by SMH110** [comment]
You can always preview your comment before submitting it 
image

**Comment by yaaHosein** [comment]
> You can always preview your comment before submitting it
> image

Ok thanks

**Comment by yaaHosein** [comment]
The screenshot proof that I was wrong 

Only the long journey worked 
But I do not know how it worked for me when I checked it first time when I told you it’s working 

The internet is slow 
I took screenshot and cut the needed part

**Comment by SMH110** [comment]
> The screenshot proof that I was wrong
> 
> Only the long journey worked
> But I do not know how it worked for me when I checked it first time when I told you it’s working
> 
> The internet is slow
> I took screenshot and cut the needed part

I asked you about the clicked operator 
how `this.currentResult` can give you the "operator" 
I think you need to read your comment about the role of the `this.currentResult`

**Comment by yaaHosein** [comment]
> > The screenshot proof that I was wrong
> > Only the long journey worked
> > But I do not know how it worked for me when I checked it first time when I told you it’s working
> > The internet is slow
> > I took screenshot and cut the needed part
> 
> I asked you about the clicked operator
> how `this.currentResult` can give you the "operator"
> I think you need to read your comment about the role of the `this.currentResult`

No it is not 
I was wrong 
But I do remember how I saw the operator when I logged this current result in opera method previously

**Comment by SMH110** [comment]
Any update on this?

**Comment by yaaHosein** [comment]
> Any update on this?

Are you asking for updates of this figure?!
Or on the last comment?!

**Comment by SMH110** [comment]
what happ

> > Any update on this?
> 
> Are you asking for updates of this figure?!
> Or on the last comment?!

I asked you what need to be done to show the operator in the history element

**Comment by yaaHosein** [comment]
Please do you asking for showing the operator in the resultElement or in the historyElement ?!
Thanks

**Comment by SMH110** [comment]
> Please do you asking for showing the operator in the resultElement or in the historyElement ?!
> Thanks

Next Task
You have implemented pushing the current result to the history array when a button clicked.
If we wanted to do something similar to the operators buttons what should we do?

what need to be done to show the operator in the history element.
for example if I click on these buttons
1 + 2 + 3

at the moment I can see this in the hisotry element
1..12..123..

what need to be done to see
1,+,.2,+,3 ?

**Comment by yaaHosein** [comment]
> > Please do you asking for showing the operator in the resultElement or in the historyElement ?!
> > Thanks
> 
> Next Task
> You have implemented pushing the current result to the history array when a button clicked.
> If we wanted to do something similar to the operators buttons what should we do?
> 
> what need to be done to show the operator in the history element.
> for example if I click on these buttons
> 1 + 2 + 3
> 
> at the moment I can see this in the hisotry element
> 1..12..123..
> 
> what need to be done to see
> 1,+,.2,+,3 ?

since some days ago I have removed the .. 
and made the history show operators and pushed the updates 
And now I can see the operators in the history div
But with errors 
I mean when I click on * 
All the operators change  to * 
Even if I had clicked on + 
If I close click on -
All the operators change to - 
Even if I was clicked on + previously 
I tried a little bit to know y this is happening but I did not reach to any solutions yet 
Thanks

**Comment by SMH110** [comment]
> > > Please do you asking for showing the operator in the resultElement or in the historyElement ?!
> > > Thanks
> > 
> > 
> > Next Task
> > You have implemented pushing the current result to the history array when a button clicked.
> > If we wanted to do something similar to the operators buttons what should we do?
> > what need to be done to show the operator in the history element.
> > for example if I click on these buttons
> > 1 + 2 + 3
> > at the moment I can see this in the hisotry element
> > 1..12..123..
> > what need to be done to see
> > 1,+,.2,+,3 ?
> 
> since some days ago I have removed the ..
> and made the history show operators and pushed the updates
> And now I can see the operators in the history div
> But with errors
> I mean when I click on *
> All the operators change to *
> Even if I had clicked on +
> If I close click on -
> All the operators change to -
> Even if I was clicked on + previously
> I tried a little bit to know y this is happening but I did not reach to any solutions yet
> Thanks

I see this when I click on a button
image

**the requirement was to show comma , not null**
also, I asked you to not touch the `gettingHistoryArray` method 
why did you pushed changes to the `gettingHistoryArray`?

File: DevelopedCalculator.js [hunk 8]
```diff
@@ -30,6 +30,8 @@ class DevelopedCalculator extends Calculator {
     onOperatorButtonClick(event) {
         // calling parent class's onOperatorButtonClick method
         super.onOperatorButtonClick(event);
+        const currentResultAndclickedOperButton = this.currentResult + event.target.dataset.operator;
+        this.currentResult = currentResultAndclickedOperButton;
         // pushing the currentResult to the history array 
         this.history.push(this.currentResult);
```
Comment (SMH110) [inline DevelopedCalculator.js]: can you tell the reason behind pushing the `currentResult` here?

**Comment by yaaHosein** [comment]
After all of these much of comments 
I did not added new codes to parent class 
Maybe some old comments only 

Thanks

File: DevelopedCalculator.js (hunk 8 above)
Comment (yaaHosein) [reply DevelopedCalculator.js]: I am pushing the the current result and clicked operator now to update the history element

File: DevelopedCalculator.js [hunk 9]
```diff
@@ -33,7 +33,7 @@ class DevelopedCalculator extends Calculator {
         const currentResultAndclickedOperButton = this.currentResult + event.target.dataset.operator;
         this.currentResult = currentResultAndclickedOperButton;
         // pushing the currentResult to the history array 
-        this.history.push(this.currentResult);
+        this.history.push(currentResultAndclickedOperButton);
```
Comment (SMH110) [inline DevelopedCalculator.js]: what is the reason of pushing the `currentResult` to the history array?

**Comment by yaaHosein** [comment]
No need to push it in onOpertsorButtonNumberClick

**Comment by SMH110** [comment]
This is good now, you can complete this PR

**Comment by yaaHosein** [comment]
Ok thanks

**Pull request merged**